#! /bin/sh /usr/share/dpatch/dpatch-run
## 63_warn_on_missing_default_fields.dpatch by Kel Modderman <kel@otaku42.de>
##
## DP: Goal: Emit warning when script contains header with missing
## DP: Default-Start or Default-Stop  LSB comment keywords in addition to
## DP: Provides, Required-Start and Required-Stop. This is consistent with
## DP: lintian warning:
## DP:   http://lintian.debian.org/reports/tags/init.d-script-missing-lsb-keyword.html
## DP:
## DP: Fixes: none
## DP:
## DP: Status: Intend to submit upstream.
## DP:
## DP: Notes: test_no_default_start and test_no_default_stop test suite cases
## DP: expose this situation.

@DPATCH@
diff -urNad insserv~/insserv.c insserv/insserv.c
--- insserv~/insserv.c	2008-05-19 23:41:03.000000000 +1000
+++ insserv/insserv.c	2008-05-19 23:44:16.094012864 +1000
@@ -1257,9 +1257,9 @@
     }
 
 #ifndef USE_STOP_TAGS
-    if (begin && end && (!provides || !required_start))
+    if (begin && end && (!provides || !required_start || !default_start))
 #else  /* USE_STOP_TAGS */
-    if (begin && end && (!provides || !required_start || !required_stop))
+    if (begin && end && (!provides || !required_start || !required_stop || !default_start || !default_stop))
 #endif /* USE_STOP_TAGS */
     {
 	char *name = basename(path);
@@ -1270,9 +1270,13 @@
 	    warn("Missing entry for Provides: please add even if empty.\n");
 	if (!required_start)
 	    warn("Missing entry for Required-Start: please add even if empty.\n");
+	if (!default_start)
+	    warn("Missing entry for Default-Start: please add even if empty.\n");
 #ifdef USE_STOP_TAGS
 	if (!required_stop)
 	    warn("Missing entry for Required-Stop: please add even if empty.\n");
+	if (!default_stop)
+	    warn("Missing entry for Default-Stop: please add even if empty.\n");
 #endif /* USE_STOP_TAGS */
     }
 
